### PR TITLE
Week 5 - Karin Nordkvist (Weatherbug)

### PR DESCRIPTION
**Pull request by karinnordkvist**
Link to my deployed project: https://weatherbug.netlify.app/

My reflections on how this week's project turned out:
I really enjoyed this week's task. It was very challenging at times, and also very rewarding as I learned to manage and modify my functions.  

Things I'd like to have clarified or explained in more detail:
The text-field and the button is not part of a form, due mainly to time limitation. I will work on this whenever I get the chance.

### CODE REVIEW

File: code/index.html [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+<body>
+
+  <header class="header">
+    <img class="header--image" src="./images/weatherbug-logo.png" />
+    <p class="header--text">WeatherBug</p>
+  </header>
+
+  <main class="main">
+    <!-- LOCATION SEARCH -->
```
Comment (AnnaHellqvist) [inline code/index.html:21]: Overall good commenting, indentation and naming of classes :) 🥇

File: code/style.css [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+
+.day--img {
+  display: inline-block;
+  flex-grow: 2;
+  text-align: right;
+  margin-right: 10px;
+  width: 4vw;
+}
+
+#day1, #day2, #day3, #day4, #day5 {
```
Comment (AnnaHellqvist) [inline code/style.css:213]: Is it possible to use the day--temp class for this styling?

File: code/script.js [hunk 3]
```diff
@@ -0,0 +1,295 @@
+//For this project I will add the api.js-file to Github. 
+// But for other projects I will put it in the .gitignore-file
+import { API_KEY } from './api.js';
+let city = 'Lerum, Sweden';
+
+// HTML QUERIES
```
Comment (AnnaHellqvist) [inline code/script.js:6]: Great job with variables for queries, makes the code cleaner further down! 👍 💯

File: code/script.js [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+const weatherTag = document.getElementById('weather');
+const sunriseTag = document.getElementById('sunrise');
+const sunsetTag = document.getElementById('sunset');
+const sunriseImg = document.getElementById('sunrise-image');
+const sunsetImg = document.getElementById('sunset-image');
+const changeLocInput = document.getElementById('location-input')
+const changeLocButton = document.getElementById('location-button')
+const changeLocGeo = document.getElementById('location-geo')
+const weekBox = document.querySelector('.week-box')
+const weatherDescriptions = [
```
Comment (AnnaHellqvist) [inline code/script.js:23]: nice approach with array of objects, I tried this myself but couldnt get it to work as I wanted it to :) 👍

File: code/script.js [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+const fetchAPILink = (selectedCity) => {
+  city = selectedCity
+  const forecastApiUrl = `https://api.openweathermap.org/data/2.5/forecast?q=${city}&units=metric&appid=${API_KEY}`;
+  fetchForecastData(forecastApiUrl)
+  const weatherApiUrl = `https://api.openweathermap.org/data/2.5/weather?units=metric&q=${city}&appid=${API_KEY}`;
+  fetchWeatherData(weatherApiUrl)
+};
+
+// I separated the fetch-function from the api-declaration to be able
+// to use the function for different types of API-requests
```
Comment (AnnaHellqvist) [inline code/script.js:86]: Good commenting! Makes it easy for me to understand!

File: code/script.js [hunk 6]
```diff
@@ -0,0 +1,295 @@
...
+const fetchWeatherData = (apiUrl) => {
+  fetch(apiUrl)
+    .then((results) => {
+      return results.json();
+    })
+    .then((weather) => {
+      cityTag.innerHTML = weather.name;
+      temperatureTag.innerHTML = roundedNumber(weather.main.temp) + ` C°`;
```
Comment (AnnaHellqvist) [inline code/script.js:94]: Nice that you use a lot of functions 👍

**Review by AnnaHellqvist — APPROVED** [review]
Great job on the weather project Karin! I really enjoyed the humor you brought to the page with the giphs! Your page works fine and is responsive. Your code is clean and the comments makes it easy to follow and understand 🥇 You really manage to include a lot from the project requirements/goals, if not all :) Be proud of yourself and keep up the good work!

File: code/api.js [hunk 7]
```diff
@@ -0,0 +1 @@
+export const API_KEY = 'ff72c94f2c17b0a54c4cd7444d1d002b';
```
Comment (HIPPIEKICK) [inline code/api.js:1]: Ohh ;)

File: code/script.js [hunk 8]
```diff
@@ -0,0 +1,295 @@
...
+      cityTag.innerText = 'Oops!'
+      weatherImage.src = 'https://media.giphy.com/media/pvO8ugi72HKww/source.gif'
+      weatherTag.innerText = 'No city here. Search again!'
+      temperatureTag.innerText = ''
+      feelsLikeBox.style.display = 'none'
+      sunriseTag.innerText = ''
+      sunsetTag.innerText = ''
+      sunriseImg.src = './images/blank.png'
+      sunsetImg.src = './images/blank.png'
+    });
```
Comment (HIPPIEKICK) [inline code/script.js:127]: 😆 ⭐

File: code/script.js [hunk 9]
```diff
@@ -0,0 +1,295 @@
...
+  fetchAPILink(city);
+  changeLocInput.value = "";
+}
+
+// Call above function with the changeLocButton.
+changeLocButton.addEventListener('click', changeLocation)
+
+// YES, It's a quickfix. Did not have time to fix a proper form. So kill me!
+changeLocInput.addEventListener('keyup', (e) => {
+  if (e.key === 'Enter') {
```
Comment (HIPPIEKICK) [inline code/script.js:186]: ⭐

File: code/script.js [hunk 10]
```diff
@@ -0,0 +1,295 @@
...
+    changeLocation()
+  }
+});
+
+// Get current longitude- and latitude values and push them into a new API-link
+// Then use the new API-link and fetch data.
+function geoLocate() {
```
Comment (HIPPIEKICK) [inline code/script.js:193]: Arrow function? 😇

File: code/script.js [hunk 11]
```diff
@@ -0,0 +1,295 @@
...
+    timestyle: 'long',
+    hour12: false,
+    hour: '2-digit',
+    minute: '2-digit',
+  });
+};
+
+// Convert tamestamp to weekday-string
+const getWeekday = (t) => {
+  const dt = new Date(t * 1000);
```
Comment (HIPPIEKICK) [inline code/script.js:226]: Hmm, can we have more descriptive names than `t` and `dt`?

File: code/script.js [hunk 12]
```diff
@@ -0,0 +1,295 @@
...
+  } else if (weather == weatherDescriptions[8].name) {
+    weatherImage.src = weatherDescriptions[8].img;
+  } else if (weather == weatherDescriptions[9].name) {
+    weatherImage.src = weatherDescriptions[9].img;
+  } else if (weather == 'Haze') {
+    weatherImage.src = weatherDescriptions[8].img;
+  } else if (weather == 'Clear') {
+    weatherImage.src = weatherDescriptions[0].img;
+  }
+};
```
Comment (HIPPIEKICK) [inline code/script.js:292]: Can you think of a way to make this shorter using the index?

File: code/style.css [hunk 13]
```diff
@@ -0,0 +1,312 @@
...
+}
+
+body {
+  font-family: 'Rubik', sans-serif;
+  margin: auto;
+  background-color: #faaca8;
+  background-image: linear-gradient(19deg, #faaca8 0%, #ddd6f3 100%);
+}
+
+
```
Comment (HIPPIEKICK) [inline code/style.css:21]: Unnecessary whitespace

File: code/style.css [hunk 14]
```diff
@@ -0,0 +1,312 @@
...
+.cold-bg {
+  background-color: #6b7db5;
+  background-image: linear-gradient(19deg, #6b7db5 0%, #ddd6f3 100%);
+}
+
+.hot-bg {
+  background-color: #ff984b;
+  background-image: linear-gradient(19deg, #ff984b 0%, #f3d6e8 100%);
+}
```
Comment (HIPPIEKICK) [inline code/style.css:31]: 🤩

**Review by HIPPIEKICK — CHANGES_REQUESTED** [review]
Good job Karin! Be consistent with arrow functions and try to keep the styling to the CSS-file (by toggling/adding/removing classes). Some more comments for the complex parts of your code would be nice as well :)

Since one of the requirements this week was that the app should be responsive - take another look at the image as it causes some horizontal scrolling on your page. Get rid of that side scroll and you'll be approved!

**Comment by karinnordkvist** [comment]
> Good job Karin! Be consistent with arrow functions and try to keep the styling to the CSS-file (by toggling/adding/removing classes). Some more comments for the complex parts of your code would be nice as well :)
> 
> Since one of the requirements this week was that the app should be responsive - take another look at the image as it causes some horizontal scrolling on your page. Get rid of that side scroll and you'll be approved!

@HIPPIEKICK Thank you for the great feedback. Big OOPS on the function not being written correctly, sorry about that. Will definitely go back into my code and look at these things. However, I can't seem to get the horizontal scrolling to happen for me, could you send a screenshot and/or let me know what screensize you experienced this? 🤩🤓🤓

**Comment by HIPPIEKICK** [comment]
> > Good job Karin! Be consistent with arrow functions and try to keep the styling to the CSS-file (by toggling/adding/removing classes). Some more comments for the complex parts of your code would be nice as well :)
> > Since one of the requirements this week was that the app should be responsive - take another look at the image as it causes some horizontal scrolling on your page. Get rid of that side scroll and you'll be approved!
> 
> @HIPPIEKICK Thank you for the great feedback. Big OOPS on the function not being written correctly, sorry about that. Will definitely go back into my code and look at these things. However, I can't seem to get the horizontal scrolling to happen for me, could you send a screenshot and/or let me know what screensize you experienced this? 🤩🤓🤓

Of course! Between 950px - 1200px the image seems to be too wide to fit to the side of the weather data :)
Skärmavbild 2020-10-13 kl  09 39 22

**Review by HIPPIEKICK — APPROVED** [review]

**Pull request closed without merging**